src/ensembl/src/content/app/browser/Browser.tsx

```diff
@@ -122,37 +105,83 @@ export const Browser: FunctionComponent<BrowserProps> = (
   const [trackStatesFromStorage, setTrackStatesFromStorage] = useState<
     TrackStates
   >({});
+  const isFirstRenderRef = useRef(true);
+  const lastGenomeIdRef = useRef(props.activeGenomeId);
+
+  const setDataFromUrl = () => {
+    const { genomeId = null } = props.match.params;
+    const { focus = null, location = null } = props.browserQueryParams;
+    const chrLocation = location ? getChrLocationFromStr(location) : null;
+
+    const lastGenomeId = lastGenomeIdRef.current;
+
+    /*
+      before committing url changes to redux:
+      - make sure we don't already have these same values in redux store;
+      - if we already have these values, it's possible that this is because
+        the user is switching back to a previously viewed species;
+        so check whether the genome id has changed from the previous render
+        (that's the reason for lastGenomeIdRef here)
+
+      TODO: after both genome browser and browser chrome are updated so that
+      we do not update url location while moving or zooming the image; we can
+      remove the genomeId === lastGenomeId check in the if-statement below
+      and move dispatchBrowserLocation(genomeId, chrLocation) above the if-statement
+    */
+    if (
+      !genomeId ||
+      (genomeId === lastGenomeId &&
+        genomeId === props.activeGenomeId &&
+        focus === props.activeEnsObjectId &&
+        isEqual(chrLocation, props.chrLocation))
+    ) {
```

Comment:
Will the second part of this IF condition be true only when we click the reset button?

Reply:
Second part of the if-statement (`(genomeId === lastGenomeId && genomeId === props.activeGenomeId && focus === props.activeEnsObjectId && isEqual(chrLocation, props.chrLocation))`) will actually be true most of the times `props.match.params.genomeId` or `props.location.search` change 😃 

The sequence of events is as follows:
- user drags browser image
- genome browser sends an event to React wrapper telling it that chromosome location has changed
- this event is captured in redux store; and chrLocation gets updated
- react wrapper sees that the chrLocation in redux store has updated and updates the url
- this causes `props.location.search` string to change. But the values encoded in the url are now the same as we keep in redux store; so this if-statement will evaluate to true; so we will not do anything.

Reply:
Cool, thanks 👍